### PR TITLE
Fix edit button and add local url configuration

### PR DESCRIPTION
**Pull request by mrglaser**
* Edit button is only shown when user has edit rights
* Add $wgDrawioEditorUrl and $wgDrawioEditorLocal for configuration of
  local drawio instances

[3.1]

ERM14536

### CODE REVIEW

File: DrawioEditor.php [hunk 1]
```diff
@@ -35,10 +37,12 @@ public static function onOutputPageParserOutput(&$outputPage, $parseroutput) {
     }
 
     public static function parse(&$parser, $name=null) {
-        global $wgUser, $wgEnableUploads;
+        global $wgUser, $wgEnableUploads, $wgTitle;
```
Comment (osnard) [inline DrawioEditor.php]: Don't use `$wgTitle` when you can use `$parser->getTitle()` for the real context.

File: DrawioEditor.php [hunk 2]
```diff
@@ -24,6 +24,8 @@
 /* Config Defaults */
 $wgDrawioEditorImageType = 'svg';
 $wgDrawioEditorImageInteractive = false;
+$wgDrawioEditorUrl = "https://www.draw.io";
```
Comment (osnard) [inline DrawioEditor.php]: I'd rather have a more descriptive config variable name. Like `$wgDrawioEditorServiceUrl` or `$wgDrawioEditorBackendUrl`

File: DrawioEditor.php [hunk 3]
```diff
@@ -113,23 +119,26 @@ public static function parse(&$parser, $name=null) {
...
+        $edit_ahref = sprintf("<a href='javascript:editDrawio(\"%s\", %s, \"%s\", %s, %s, %s, %s, \"%s\", %s)'>",
             $id,
             json_encode($img_name, JSON_HEX_QUOT | JSON_HEX_APOS),
             $opt_type,
             $opt_interactive ? 'true' : 'false',
             $opt_height === 'chart' ? 'true' : 'false',
             $opt_width === 'chart' ? 'true' : 'false',
-	    $opt_max_width === 'chart' ? 'true': 'false');
-        
+	    $opt_max_width === 'chart' ? 'true': 'false',
```
Comment (osnard) [inline DrawioEditor.php]: Bad indention. This file does not use tabs at the beginning of the line, but spaces

File: DrawioEditor.php [hunk 4]
```diff
@@ -24,6 +24,8 @@
 /* Config Defaults */
 $wgDrawioEditorImageType = 'svg';
 $wgDrawioEditorImageInteractive = false;
+$wgDrawioEditorUrl = "https://www.draw.io";
+$wgDrawioEditorLocal = false;
```
Comment (osnard) [inline DrawioEditor.php]: Why are we creating a dedicated configuration here? Couldn't we just assume that if it is not the original service URL, it's a local URL?

E.g. `var localAttr = this.url !== 'https://www.draw.io' ? "&local=1" : "";`

**Review by osnard — CHANGES_REQUESTED** [review]

File: DrawioEditor.php (hunk 1 above)
Comment (mrglaser) [reply DrawioEditor.php]: fixed

File: DrawioEditor.php (hunk 2 above)
Comment (mrglaser) [reply DrawioEditor.php]: fixed

File: DrawioEditor.php (hunk 3 above)
Comment (mrglaser) [reply DrawioEditor.php]: fixed

File: DrawioEditor.php (hunk 4 above)
Comment (mrglaser) [reply DrawioEditor.php]: fixed

File: DrawioEditor.php [hunk 5]
```diff
@@ -48,12 +50,14 @@ public static function parse(&$parser, $name=null) {
...
 	}
-       
+
         $opt_type = array_key_exists('type', $opts) ? $opts['type'] : $wgDrawioEditorImageType;
         $opt_interactive = array_key_exists('interactive', $opts) ? true : $wgDrawioEditorImageInteractive;
         $opt_height = array_key_exists('height', $opts) ? $opts['height'] : 'auto';
         $opt_width = array_key_exists('width', $opts) ? $opts['width'] : '100%';
         $opt_max_width = array_key_exists('max-width', $opts) ? $opts['max-width'] : false;
+        $opt_url = array_key_exists('url', $opts) ? $opts['url'] : $wgDrawioEditorBackendUrl;
+        $opt_local = ($wgDrawioEditorBackendUrl === "https://www.draw.io");
```
Comment (osnard) [inline DrawioEditor.php]:
```suggestion
        $opt_local = ($wgDrawioEditorBackendUrl !== "https://www.draw.io");
```

**Review by osnard — CHANGES_REQUESTED** [review]

**Pull request merged**